### PR TITLE
opencl-icd-loader: add 2022.05.18 + conan v2 support

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — APPROVED** [review]

File: recipes/opencl-icd-loader/all/conanfile.py [hunk 1]
```diff
@@ -1,31 +1,35 @@
-from conans import ConanFile, CMake, tools
+from conan import ConanFile, conan_version
```
Comment (jwillikers) [inline recipes/opencl-icd-loader/all/conanfile.py:1]: Sounds like you're supposed to remove the `conan_version` here.

**Review by jwillikers — CHANGES_REQUESTED** [review]

**Review by jwillikers — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

File: recipes/opencl-icd-loader/all/conanfile.py (hunk 1 above)
Comment (prince-chrismc) [reply recipes/opencl-icd-loader/all/conanfile.py]: Thank you for catching it :)

**Review by jwillikers — APPROVED** [review]

File: recipes/opencl-icd-loader/all/conanfile.py [hunk 2]
```diff
@@ -35,49 +38,70 @@ def config_options(self):
...
-        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
-        cmake = self._configure_cmake()
+        copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
         cmake.install()
+        rm(self, "*.pdb", os.path.join(self.package_folder, "bin"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
 
     def package_info(self):
+        self.cpp_info.set_property("cmake_find_mode", "both")
```
Comment (uilianries) [inline recipes/opencl-icd-loader/all/conanfile.py:91]: Why cmake_find_mode:both is required here?

Comment (SpaceIm) [reply recipes/opencl-icd-loader/all/conanfile.py:91]: https://cmake.org/cmake/help/latest/module/FindOpenCL.html

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**